script.js

```diff
@@ -1 +1,77 @@
+$(document).ready(function () {
+    $(document).on("scroll", onScroll);
+    $(window).on("popstate", onBackForward);
+    
+    //smoothscroll
+    $('a[href^="#"]').on('click', function (e) {
+        e.preventDefault();
+        $(document).off("scroll");
+        
+        
+        $('a').parent().each(function () {
+            $(this).removeClass('active');
+        })
+        
+        $(this).parent().addClass('active');
+        
+        var target = this.hash,
+            menu = target;
+        $target = $(target);
+        
+        // also add link to history when it is clicked on
+        if (history.pushState) {
+            history.pushState({url: target}, null, target);
+        }
+        
+        $('html, body').stop().animate({
+            'scrollTop': $target.offset().top
+        }, 500, 'swing', function () {
+            window.location.hash = target;
+            $(document).on("scroll", onScroll);
```

Comment:
Why do you attach this again here? You have a handler for `scroll` events on line 7

Reply:
it gets detached on line 8